src/raftstore/store/peer.rs

```diff
@@ -459,13 +462,42 @@ impl Peer {
             ready.hs.take();
         }
 
+        if let Some(ref soft_state) = ready.ss {
+            if soft_state.leader_id == raft::INVALID_ID {
+                if self.leader_missing_time.is_none() {
+                    self.leader_missing_time = Some(Instant::now())
+                }
+            } else {
+                // When a peer is not initialized, it has no data at storage.
+                // Even if a leader sends heartbeats to it, we consider it as
+                // in the `leader missing` state. That is because if it's isolated from the leader
+                // before it could successfully receive snapshot from the leader and
+                // apply that snapshot, no raft ready event will be triggered,
+                // so that we could not detect the leader is missing for it at here.
```

Comment:
> If the configuration is valid (pass the validate check), then the group should work in some way(good performance or bad performance).

How does the configuration validate check work against the example I mentioned above?

> Besides, even the region is not working for hours, it won't cause any problem

Part of it is true. But in a long-playing system which runs for years, there probably be plenty of these zombie peers, they consume resources to be managed. It's better that we could target them and destroy them at some "right" way.